New multiple thumbnail sample

Fedes365 opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New sample?      | yes
| Related issues?  | no related issues

#### What's in this Pull Request?

A new column sample to create a multiple thumbnail, through images URLs listed in a multiple choice column.

Fedes365 commented:
Direct link here: https://github.com/Fedes365/List-Formatting/tree/master/column-samples/multiple-thumbnail

Fedes365 commented:
@tecchan1107 

I made some changes to remove any limitation related to the suggested apps.

I tried also to illustrate a use case with a lookup column instead of a multi-choice, when SharePoint images are stored in a document library (of the same site). However, it seems that it's not possible, because a lookup column cannot have a document library file name column as a data source 😱... 😓

Therefore I tried a workaround by using also a calculated column as intermediary data source (basically a "waterfall" of data source), but I realized a calculated column couldn't recognize the OOTB file name column as existing.

Well, this is the maximum I was able to build by playing with URLs arrays. So... ready to review.

Thanks and see you later! 👋

tecchan1107 commented:
Thank you for the excellent and beautiful sample @Fedes365 🙏✨
I had an opportunity to display logos and used this sample as soon as possible! It is very easy to understand if it is displayed in an image rather than in text!

After using the sample, I had three concerns.

1. The first thing that concerned me was that the images were cut off unless the column width was widened for both basic and advance.
![image](https://github.com/pnp/List-Formatting/assets/51254139/1e8f9e60-de9a-432f-bcf8-099dcd499712)
How about removing `"min-width": "730px"` and modifying the image to be wrapped to fit the column width?
![sample](https://github.com/pnp/List-Formatting/assets/51254139/bb6d9754-ee40-4847-ab27-e3ad2fcc2c92)

2. Regarding advance, sorry I can't really put it into words, but I feel uncomfortable updating the column values to enlarge the image... If the purpose is to display images in a larger size, how about displaying them in a custom card, like [image-lightbox](https://github.com/pnp/List-Formatting/tree/master/column-samples/image-lightbox)? This way, there is no need to update the columns.

3. Regarding the link to Microsoft Designer in advance, I was confused as to what to do after clicking on the link... If a link to Microsoft Designer was to be displayed, I thought it would be better if there was an note in the screen, or a reason to display the link in the README, etc.

Fedes365 commented:
Hello @tecchan1107 and thanks a lot for your feedback! I will touch all the points you reported:

**1)** Yes, the column has to be widened because its min-width is set to 730px. This could be properly explained in the README, so users would know in advance what to expect and what they would need to do.
I put a min-width of 730px to overcome a "by design" limitation of the OOTB form. I could remove that min-width value, but here is how the user interface would be in the OOTB SharePoint form (same result in Lists app):

![Form-nowidth-res](https://github.com/pnp/List-Formatting/assets/95828495/bca1b4fa-f2ee-42dc-8957-487f72981f80)

I have the suspect that it's by design, because of the REACT container in the OOTB form having a max-width of 256px and therefore it doesnt adapt itself based on the underlying JSON formatting.

![Form-width-core](https://github.com/pnp/List-Formatting/assets/95828495/6015d5ce-3651-408e-b2cf-962504a38fca)

I believe this app limitation cannot be removed nor it can be properly explained to users, since the resulting form interface would look pretty weird. However, I could also be wrong and if you find a way to improve or solve it through JSON, we can update the pull request 👍

**2)** Yes, I tried to create an image lightbox for each image displayed, by taking that solution as a reference. However, from my testing, custom hovercards completely ignore the underlying **_"forEach": "choiceIterator in @currentField"_**. In other words, each hovercard was reproducing the entire array of images, so behaving like an autonomous div, completely isolated from the surrounding context. Probably this is "by design".
However, if you find a way to solve this limitation, we could update the README and replace with a lightbox 👍

Alternatively, I could also explain why a lightbox is not supported or even remove the advanced sample.

**3)** Regarding Microsoft Designer, I completely agree with you. I thought it could be useful to edit and reuse any image displayed in a list, but after thinking twice, I realized that the Designer app is still a personal tool and, in my opinion, it doesn't make sense to keep that link in an enterprise environment at the moment. Therefore, I could update the JSON code to remove that icon and its corresponding link 👌


Thanks again to help me improve this sample! 👍💙

tecchan1107 commented:
Hi, @Fedes365 !

> 1) Yes, the column has to be widened because its min-width is set to 730px. This could be properly explained in the README, so users would know in advance what to expect and what they would need to do.
I put a min-width of 730px to overcome a "by design" limitation of the OOTB form. I could remove that min-width value, but here is how the user interface would be in the OOTB SharePoint form (same result in Lists app):

I see! Thank you for the very clear explanation and screenshots!!
I tried to find a solution, but at this point I couldn't figure it out...

> 2) Yes, I tried to create an image lightbox for each image displayed, by taking that solution as a reference. However, from my testing, custom hovercards completely ignore the underlying "forEach": "choiceIterator in @currentField". In other words, each hovercard was reproducing the entire array of images, so behaving like an autonomous div, completely isolated from the surrounding context. Probably this is "by design".
However, if you find a way to solve this limitation, we could update the README and replace with a lightbox 👍
Alternatively, I could also explain why a lightbox is not supported or even remove the advanced sample.

I tried it and it is true that I could not get the foreach value from customCard! I didn't know this😱😱
If you include a description of how you were unable to implement the lightbox, someone who sees this sample later and finds a solution may make a pull request. Also, it may prevent the creation of an issue. So, I think it will be a better sample if it is written.

> 3) Regarding Microsoft Designer, I completely agree with you. I thought it could be useful to edit and reuse any image displayed in a list, but after thinking twice, I realized that the Designer app is still a personal tool and, in my opinion, it doesn't make sense to keep that link in an enterprise environment at the moment. Therefore, I could update the JSON code to remove that icon and its corresponding link 👌

I, too, thought it would be very useful to have an image editing window open! However, Microsoft Designer does not seem to currently have that feature...let's hope they add it in the future!
I am sorry... but may I ask you to remove the Microsoft Designer link?

Fedes365 commented:
Hi @tecchan1107!

- Updated the advanced sample JSON to remove the obsolete brush icon
- Added a short and visual explanation about column min-width, how to expand it and how to save current settings based on Microsoft official support reference
- Added a short explanation about current Microsoft Lists limitation, when it comes to images array and custom hovercards.

Please, let me know if further changes are required.

Thanks for your support!

tecchan1107 commented:
Merged! Thank you for the beautiful sample and for the fixes🙏✨
And sorry for the delay🙇‍♂
